src/main/kotlin/org/jetbrains/research/kotlin/mpp/inference/operators/activations/Activation.kt

```diff
@@ -0,0 +1,31 @@
+package org.jetbrains.research.kotlin.mpp.inference.operators.activations
+
+import org.jetbrains.research.kotlin.mpp.inference.operators.Operator
+import org.jetbrains.research.kotlin.mpp.inference.tensors.Tensor
+import java.math.BigDecimal
+import kotlin.math.exp
+
+@Suppress("UNCHECKED_CAST")
+sealed class Activation<T : Number>(private val func: (T) -> T) : Operator<T>() {
+    override fun apply(inputs: Collection<Tensor<T>>): Collection<Tensor<T>> {
+        val toActivate = inputs.singleOrNull()
+        requireNotNull(toActivate) { "Multiple inputs are not allowed" }
+
+        return listOf(toActivate.mapElements(func))
+    }
+
+    class Identity<T : Number> : Activation<T>(func = { x -> x })
```

Comment:
If it has no parameters -- why should it be class? probably `object`?

Reply:
Since this class has a type parameter, we can't use `object` here